rust_snuba/src/processors/uptime_monitor_checks.rs

```diff
@@ -0,0 +1,157 @@
+use crate::config::ProcessorConfig;
+use anyhow::Context;
+use chrono::DateTime;
+use sentry_arroyo::backends::kafka::types::KafkaPayload;
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use crate::types::{InsertBatch, KafkaMessageMetadata};
+
+pub fn process_message(
+    payload: KafkaPayload,
+    metadata: KafkaMessageMetadata,
+    _config: &ProcessorConfig,
+) -> anyhow::Result<InsertBatch> {
+    let payload_bytes = payload.payload().context("Expected payload")?;
+    let (rows, origin_timestamp) =
+        deserialize_message(payload_bytes, metadata.partition, metadata.offset)?;
+
+    InsertBatch::from_rows(rows, DateTime::from_timestamp(origin_timestamp as i64, 0))
+}
+
+pub fn deserialize_message(
+    payload: &[u8],
+    partition: u16,
+    offset: u64,
+) -> anyhow::Result<(Vec<UptimeMonitorCheckRow>, f64)> {
+    let monitor_message: UptimeMonitorCheckMessage = serde_json::from_slice(payload)?;
+
+    let rows = vec![UptimeMonitorCheckRow {
+        organization_id: 1,
+        project_id: 1,
+        environment: monitor_message.environment,
+        uptime_subscription_id: monitor_message.subscription_id,
+        uptime_check_id: monitor_message.guid,
+        scheduled_check_time: monitor_message.scheduled_check_time_ms,
+        timestamp: monitor_message.actual_check_time_ms,
+        duration: monitor_message.duration_ms,
+        region_slug: Some("global".to_string()),
+        check_status: monitor_message.status,
+        check_status_reason: monitor_message.status_reason.map(|r| r.r#type),
+        http_status_code: monitor_message
+            .request_info
+            .unwrap()
```

Comment:
It panics if `unwrap()` fails here.

Add the `Default` trait to your `RequestInfo` structure and use `unwrap_or_default()` for both.